application/classes/Ushahidi/Repository/Form.php

```diff
@@ -83,9 +83,12 @@ public function update(Entity $entity)
             $key === 'name' ? $this->emit($this->event, $user->email, ['primary_survey_name' => $val]) : null;
           }
         }
-
+        $form = $entity->getChanged();
+        $form['updated'] = time();
+        // removing tags from form before saving
+        unset($form['tags']);
```

Comment:
I already made `tags` part of the immutable fields on the object. I'm not sure why we still need to do this. I couldn't figure out how to reproduce it locally.